### PR TITLE
Verification link delivery to email

### PR DESCRIPTION
**Pull request by saltanovas**
Service is finally working. Hope without any bugs. Take a note that verification link is delivered to email from which it was sent. Since we don't have our app email, **please add new values to your secrets** :

"Email": {
    "Address": {SenderEmailAddress}
    "Password": {SenderEmailAddressPassword}
  }

### CODE REVIEW

File: SmartSaver.Domain/Services/EmailServices/Mailer.cs [hunk 1]
```diff
@@ -1,80 +1,36 @@
...
-
-                using (var client = new SmtpClient())
-                {
-                    client.ServerCertificateValidationCallback = (s, c, h, e) => true;
-                    client.Connect(_smtpSettings.Server, _smtpSettings.Port, false);
-                    client.Authenticate(_smtpSettings.Username, _smtpSettings.Password);
-                    client.Send(message);
-                    client.Disconnect(true);
-                }
+                _smtp.SendAsync(message, "");
```
Comment (povilasleka) [inline SmartSaver.Domain/Services/EmailServices/Mailer.cs:26]: If you are using async method, it should be awaited.

File: SmartSaver.Domain/Services/EmailServices/Mailer.cs [hunk 2]
```diff
@@ -1,80 +1,36 @@
...
-            {
-                throw new InvalidOperationException(e.Message);
-            }
+                Credentials = new NetworkCredential() { UserName = _configuration["Email:UserName"], Password = _configuration["Email:Password"] }
+            };
+            _smtp.EnableSsl = true;
         }
 
-        public void SendEmail(string email, string subject, string body)
+        public virtual MailStatus SendEmailAsync(MailMessage message)
```
Comment (povilasleka) [inline SmartSaver.Domain/Services/EmailServices/Mailer.cs:22]: Should be async method.

File: SmartSaver.Domain/Services/EmailServices/Mailer.cs (hunk 1 above)
Comment (saltanovas) [reply SmartSaver.Domain/Services/EmailServices/Mailer.cs:26]: Maybe it is good not to await for it in this particular case?

Comment (povilasleka) [reply SmartSaver.Domain/Services/EmailServices/Mailer.cs:26]: Sorry for making a mistake, SmtpClient.SendAsync method does not block the calling thread and is type void, therefore it shouldn't be used with an await keyword.

Comment (saltanovas) [reply SmartSaver.Domain/Services/EmailServices/Mailer.cs:26]: Got it!

**Pull request merged**